azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageTypeDaoImpl.java

```diff
@@ -153,8 +179,16 @@ public int createImageType(final ImageType imageType) {
       imageTypes = this.databaseOperator
           .query(FetchImageTypeHandler.FETCH_IMAGE_TYPE_BY_NAME, fetchImageTypeHandler,
               name.toLowerCase());
+      if (imageTypes == null) {
+        log.error("Unable to fetch image type metadata from image type : " + name);
+        throw new ImageMgmtDaoException(ErrorCode.NOT_FOUND,
+            "Unable to fetch image type metadata for image type : " + name);
+      }
       // Check if there are more then one image types for a given name. If so throw exception
       if (imageTypes != null && imageTypes.size() > 1) {
+        log.error(
+            "Failed to get image type with ownerships by name. Can't have more that one image type"
+                + " record for a given type with name : " + name);
         throw new ImageMgmtDaoException(ErrorCode.NOT_FOUND, "Failed to get image type with "
             + "ownerships by name. Can't have more that one image type record for a given type "
             + "with name : " + name);
```

Comment:
In `getImageTypeWithOwnershipsById` we returned `ErrorCode.UNPROCESSABLE_ENTITY` in this case. Can we agree on one status and be consistent?

Reply:
Thanks. I think we should just go with ErrorCode.UNPROCESSABLE_ENTITY. I will change it.